### PR TITLE
MULE-17401: Scheduler source should route emitted message through FlowProcessMediator

### PR DESCRIPTION
**Pull request by plbalbi**

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/internal/source/scheduler/DefaultSchedulerMessageSource.java [hunk 1]
```diff
@@ -157,37 +162,30 @@ private void poll() {
...
-          .doOnError(MessagingException.class,
-                     me -> ((BaseEventContext) me.getEvent().getContext()).error(me))
-          .doOnSuccess(result -> ((BaseEventContext) result.getContext()).success())
-          .doFinally(s -> {
-            synchronized (DefaultSchedulerMessageSource.this) {
-              executing = false;
-            }
-          })
-          .subscribe(requestUnbounded());
+      setCurrentEvent(InternalEvent.builder(EventContextFactory.create(flowConstruct, getLocation())).message(Message.of(null))
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/source/scheduler/DefaultSchedulerMessageSource.java]: review if this is needed or already done by the ModuleFlowProcessingPhase

File: core/src/main/java/org/mule/runtime/core/internal/source/scheduler/SchedulerFlowProcessingTemplate.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  @Override
+  public Publisher<CoreEvent> routeEventAsync(Publisher<CoreEvent> eventPub) {
+    return from(eventPub).transform(messageProcessor);
+  }
+
+  @Override
+  public Publisher<Void> sendResponseToClient(CoreEvent response, Map<String, Object> parameters) {
+    return Mono.empty();
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/source/scheduler/SchedulerFlowProcessingTemplate.java]: static import

File: core/src/main/java/org/mule/runtime/core/internal/source/scheduler/SchedulerFlowProcessingTemplate.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  @Override
+  public Publisher<Void> sendFailureResponseToClient(MessagingException exception, Map<String, Object> parameters) {
+    return Mono.empty();
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/source/scheduler/SchedulerFlowProcessingTemplate.java]: static import

File: core/src/main/java/org/mule/runtime/core/internal/source/scheduler/DefaultSchedulerMessageSource.java (hunk 1 above)
Comment (plbalbi) [reply core/src/main/java/org/mule/runtime/core/internal/source/scheduler/DefaultSchedulerMessageSource.java]: It's not done in ModuleFlowProcessingPhase. The current event is set in the interceptors applied to each processor. In my opinion is needed here.

**Pull request merged**